duckduckgo/iOS
# Lazily load tabs when launching and clear excess tabs on low memory

Reviewer: Chris
Asana: https://app.asana.com/0/414235014887631/442414558181570

**Description**:
This PR lazily loads persisted tabs on launch. It also clears loaded tabs when memory gets low.

**Steps to test this PR**:
1. In a simulator open 4 tabs
2. Background the app and kill it
3. Relaunch
4. Go to the tab switcher and select all the non-active tabs. They will each be created from the cache on demand printing a "Tab not in memory, creating" log as they are selected.
5. Switch between them, now that they are in memory there should be no new creation logs
6. Trigger a low memory warning. Half the tabs (the two OLDEST) will be removed from the cache. Selecting these will again cause the "Tab not in cache, creating" log to be displayed

###### Reviewer Checklist:

###### PR DRI Checklist:
